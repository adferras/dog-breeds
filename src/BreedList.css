.breeds {
  display: flex; }

.breed-image {
  height: 50vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-content: center; }
  .breed-image__container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 50%; }
  .breed-image img {
    max-height: 90%; }

.breed-list {
  width: 30rem; }

button {
  background: #4fbdf4;
  border-radius: 1.25rem;
  border-width: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-size: 1.5rem;
  line-height: 1.5rem;
  margin: 2px;
  outline: none;
  padding: 0 .5rem;
  text-decoration: none;
  text-transform: capitalize;
  transition: box-shadow 90ms ease-in-out, transform 90ms ease-in-out;
  width: 200px; }
  button:hover, button:focus {
    box-shadow: inset 0 -1px rgba(0, 0, 0, 0.2); }
  button:active {
    box-shadow: inset 0 1px rgba(0, 0, 0, 0.2), inset 0 100px rgba(0, 0, 0, 0.1);
    transform: translate(0, 1px); }
  button .refresh-image {
    width: 40%;
    margin: 20px; }
