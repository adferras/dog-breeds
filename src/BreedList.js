import React, { Component } from 'react';

class BreedList extends Component {
  constructor() {
    super();
    this.state = {
      breeds: [],
      breedImage: null
    };
  }

  componentDidMount() {
    fetch('https://dog.ceo/api/breeds/list/all')
      .then(results => {
        return results.json();
      })
      .then(data => {
        let breeds = Object.keys(data.message).map((breed) => {
          return (
            <button key={breed} onClick={() => this.handleClick(breed)}>
              <span>{breed}</span>
            </button>
          )
        })
        this.setState({breeds: breeds})
      })
  }

  handleClick(breed) {
    fetch(`https://dog.ceo/api/breed/${breed}/images/random`)
      .then(results => {
        return results.json();
      })
      .then(data => {
        this.setState({breedImage: data.message})
      })
  }

  render() {
    return (
      <div className="Breeds">
        <div className="breed-list">
          {this.state.breeds}
        </div>
        <img src={this.state.breedImage} alt="breed" className="breed-image" />
      </div>
    );
  }
}

export default BreedList;
