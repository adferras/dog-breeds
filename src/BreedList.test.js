import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import BreedList from './BreedList';

describe(BreedList, () => {
  const component = shallow(<BreedList />);

  it("renders and matches our snapshot", () => {
    const component = renderer.create(<BreedList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
