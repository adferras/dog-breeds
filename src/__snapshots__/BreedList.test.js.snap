// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class BreedList extends _react.Component {
  constructor() {
    super();
    this.state = {
      breeds: [],
      breedImage: null,
      breed: '' };

  }

  componentDidMount() {
    fetch('https://dog.ceo/api/breeds/list/all').
    then(results => {
      return results.json();
    }).
    then(data => {
      let breeds = Object.keys(data.message).map(breed => {
        return (
          _react2.default.createElement('button', { key: breed, onClick: () => this.handleClick(breed), __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this },
            _react2.default.createElement('span', { __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }, breed)));


      });
      this.setState({ breeds: breeds });
    });
  }

  handleClick(breed) {
    fetch(\`https://dog.ceo/api/breed/\${breed}/images/random\`).
    then(results => {
      return results.json();
    }).
    then(data => {
      this.setState({ breedImage: data.message, breed });
    });
  }

  render() {
    return (
      _react2.default.createElement('div', { className: 'breeds', __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this },
        _react2.default.createElement('div', { className: 'breed-list', __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this },
          this.state.breeds),

        this.state.breed ?
        _react2.default.createElement('div', { className: 'breed-image__container', __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this },
          _react2.default.createElement('div', { className: 'breed-image', __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this },
            _react2.default.createElement('img', { src: this.state.breedImage, alt: 'breed', __source: { fileName: _jsxFileName, lineNumber: 50 }, __self: this })),

          _react2.default.createElement('button', { className: 'refresh-image', onClick: () => this.handleClick(this.state.breed), __source: { fileName: _jsxFileName, lineNumber: 52 }, __self: this }, 'Get a new picture')) :

        null));



  }} renders and matches our snapshot 1`] = `
<div
  className="breeds"
>
  <div
    className="breed-list"
  />
</div>
`;
